esx_service/vmdk_ops.py

```diff
@@ -950,20 +965,18 @@ def findDeviceByPath(vmdk_path, vm):
         ds, disk_path = d.backing.fileName.rsplit("]", 1)
         datastore = ds[1:]
         backing_disk = disk_path.lstrip()
-        logging.debug("findDeviceByPath: datastore=%s, backing_disk=%s", datastore, backing_disk)
-
-        # Construct the parent dir and vmdk name, resolving
-        # links if any.
-        dvol_dir = os.path.dirname(vmdk_path)
-        datastore_prefix = os.path.realpath(os.path.join("/vmfs/volumes", datastore)) + '/'
-        real_vol_dir = os.path.realpath(dvol_dir).replace(datastore_prefix, "")
-        virtual_disk = os.path.join(real_vol_dir, os.path.basename(vmdk_path))
-        logging.debug("dvol_dir=%s datastore_prefix=%s real_vol_dir=%s", dvol_dir, datastore_prefix,real_vol_dir)
-        logging.debug("backing_disk=%s virtual_disk=%s", backing_disk, virtual_disk)
-        if virtual_disk == backing_disk:
-            logging.debug("findDeviceByPath: MATCH: %s", backing_disk)
-            return d
-    return None
+        backing_disk_name = vmdk_utils.strip_vmdk_extension(os.path.basename(backing_disk))
+
+        logging.debug("findDeviceByName: datastore = %s, backing_disk_name = %s",
+                        datastore, backing_disk_name)
+
+        if backing_disk_name == vol_name:
```

Comment:
@msterin 
Fixing it to compare both volume name and datastore name with backing disk name and backing disk datastore.
Confirmed that we always get a qualified volume name i.e. volumename@datastore; from docker -> vmdk_plugin -> esx host.